app/src/common/shared/com/igalia/wolvic/ui/widgets/settings/DeveloperOptionsView.java

```diff
@@ -134,17 +141,38 @@ protected void updateUI() {
         if (BuildConfig.DEBUG && mBinding.webglOutOfProcessSwitch.isChecked() != SettingsStore.WEBGL_OUT_OF_PROCESS) {
             setWebGLOutOfProcess(SettingsStore.WEBGL_OUT_OF_PROCESS, true);
             restart = true;
+            restartWebGLOutOfProcess = true;
         }
 
         if (mBinding.localAddonSwitch.isChecked() != SettingsStore.LOCAL_ADDON_ALLOWED) {
             setLocalAddon(SettingsStore.LOCAL_ADDON_ALLOWED, true);
         }
 
         if (restart) {
-            showRestartDialog();
+            showRestartDialog(cancelReset(restartDebugLogging, restartUIHardwareAcceleration, restartWebGLOutOfProcess));
         }
     };
 
+    private RestartDialogWidget.CancelFunction cancelReset(boolean restartDebugLogging,
```

Comment:
The problem with this approach is that it isn't very extensible. For each new setting we should have to add a new parameter to the method. I think we could get rid of the `restart` variable and just call directly the showRestartDialog method passing a different cancel callback

Reply:
I made it more extensible. Please check if the latest change is what you had in mind.